packages/origin-backend/src/pods/device/device.service.ts

```diff
@@ -240,20 +240,15 @@ export class DeviceService {
             throw new NotFoundException(StorageErrors.NON_EXISTENT);
         }
 
-        device.automaticPostForSale = update.automaticPostForSale;
-        if (update.automaticPostForSale) {
-            device.defaultAskPrice = update.defaultAskPrice;
-        }
-
         try {
-            await this.repository.save((device as unknown) as DeepPartial<Device>);
+            await this.repository.update(id, update);
```

Comment:
Please change this to an explicit 

{
automaticPostForSale: update.automaticPostForSale
...
} 

accepting input directly is potential security issue, as update can contain more data, malicious one like "owner".

Reply:
Available properties are defined in the type: DeviceSettingsUpdateData

Reply:
But yeah, it might still happen.